arroyo-df/src/lib.rs

```diff
@@ -821,16 +839,20 @@ impl TreeNodeRewriter for QueryToGraphVisitor {
                     )?,
                     key_fields: (0..key_count).collect(),
                 };
+
                 let aggregate_index = self.local_logical_plan_graph.add_node(
                     LogicalPlanExtension::AggregateCalculation(aggregate_calculation),
                 );
+
                 let table_name = format!("{}", aggregate_index.index());
                 self.local_logical_plan_graph.add_edge(
                     key_index,
                     aggregate_index,
                     DataFusionEdge {
-                        value_schema: input.schema().clone(),
-                        key_schema: Some(key_schema),
+                        schema: input_df_schema,
+                        edge_type: LogicalEdgeType::Shuffle,
+                        // start at 1 to exclude the window
```

Comment:
I'm not sure I understand this? I think the window field can appear anywhere in the group by list, not at the beginning. The approach I'd taken was to remove the window_field from the list and track where it should be inserted after the aggregate is computed.

Reply:
Ok, I'd misread the code here. I'll make it more general.